includes/class-database-cache.php

```diff
@@ -83,6 +83,49 @@ class Database_Cache implements MultiCurrencyCacheInterface {
 	 */
 	private $refresh_disabled;
 
+	/**
+	 * The maximum size in bytes of a cache entry that will be stored in memory.
+	 */
+	const IN_MEMORY_CACHE_ENTRY_SIZE_LIMIT = 15 * KB_IN_BYTES;
+
+	/**
+	 * The maximum data size in bytes of all the in-memory cache stored data.
+	 *
+	 * This will ensure we keep the memory impact under control.
+	 * This is especially important for large stores that can handle many concurrent requests.
+	 */
+	const IN_MEMORY_CACHE_SIZE_LIMIT = 100 * KB_IN_BYTES;
+
+	/**
+	 * In-memory cache for the duration of a single request.
+	 *
+	 * This is used to avoid multiple database reads for the same data and as a backstop in case the database write fails,
```

Comment:
Cruel nitpick: It's the first time I've come across the term 'backstop'. IMO, 'fallback' is a more common term and we should use it here.

Reply:
Nitpick reply: "[Backstop](https://www.google.com/search?q=backstop+meaning)" and "[fallback](https://www.google.com/search?q=fallback+meaning)" are not perfect synonyms. In our case, we always "pass through" the in-memory cache (as opposed to falling back on it), and, in certain cases, we don't move past it (hence being a backstop). 

A true fallback is the system we have in core for Payment Gateways Recommendations: if we are allowed to fetch remotely, we will; but if we are not allowed or fail to, we fall back to using the hard-coded ones (defaults).

WDYT? 😜

Reply:
Before (when the in-memory cache was used conditionally based on DB write errors), I thought it was a fallback to the WP options cache (not the WP cache object) in most cases.

After (now that in-memory cache is always used), 'backstop' feels more accurate. But that also means we're introducing a potentially unfamiliar term, so it's worth clarifying somewhere. Unless it's just me :)